osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -197,7 +197,7 @@ private double getScore(ScoringMode mode)
         {
             return GetScore(mode, maxHighestCombo,
                 maxBaseScore > 0 ? baseScore / maxBaseScore : 0,
-                maxHighestCombo > 0 ? (double)HighestCombo.Value / maxHighestCombo : 0,
+                maxHighestCombo > 0 ? (double)HighestCombo.Value / maxHighestCombo : 1,
```

Comment:
``maxHighestCombo`` is a bad name for this value. Should be something like ``maxAchievableCombo`` or should be documented as such

Reply:
is there something stopping you from doing either?

Reply:
Because I was unsure if this is what this variable stores exactly because it's undocumented. If I was already changing names, I would probably change ``Reset(bool storeResults)`` to ``Reset(bool storePerfectResults)`` as-well, because it seems to only be used after autoplay to get all the maximum values.

Reply:
well, you've already inferred what the variable is by suggesting the rename, right?

also, disagree on the `Reset` rename. the fact that it's used to store maximum for autoplay is incidental.

Reply:
Then there needs to be some other way to get maximum achievable combo after applying a beatmap. The ``reset`` method is only ever used in there, in ``ApplyBeatmap``, though, exactly for getting all the maximum values by simulating a perfect play.

Reply:
why?

Reply:
> the fact that it's used to store maximum for autoplay is incidental

it doesn't look incidental do me. It looks like it's the only thing it's used for (I only see reset used when beatmap is applied). If the storing of maximum score is indeed incidental, *then* there needs to be another way to store maximum values.

Reply:
I meant "incidental" in the deeper sense that it doesn't *matter* why the score processor is storing results. It just can, for future use. There's no reason to make a bigger deal out of it than it needs to be.

Reply:
I get that, but the fact remains that after the results are stored they are expected to be stored from a perfect simulation. If something else were to call the reset function to store results that aren't perfect, score calculation for that ScoreProcessor would be inaccurate

Reply:
I don't really want to continue belaboring this point as it is insignificant in my view. One look at the body of `Reset()` says what the parameter does.

If we were to *really* care that nobody should be able to store non-perfect results then the entire calculation of perfect results should be encapsulated entirely. You're welcome to try to implement that, if you care. I kind of don't - there's hundreds of more important issues open right now. Until this becomes an issue, we can deal with what we have.